File: python/fast.py

import csv
import json
 
def make_json(csvFilePath):
     
    csv_file = open(csvFilePath, "r")
    csv_reader = csv.DictReader(csv_file)
    row = 0
    for lines in csv_reader:
        output = json.dumps(lines)
        output_json = open("../data/python/json"+str(row)+".json", "w")
        output_json.write(output)
        row += 1
    output_json.close()
    csv_file.close()

csvFilePath = r'../data/data.csv'
make_json(csvFilePath)

File: ruby/fast.rb
require "csv"
require "json"
require "fileutils"

csv_path = File.join(__dir__, "../data/data.csv")
json_path = File.join(__dir__, "../data/ruby/json")
FileUtils.mkdir_p(json_path)

File.open(csv_path) do |csv_file|
  CSV.foreach(csv_file).with_index do |row, i|
    json_file_path = File.join(json_path, "json#{i}.json")
    File.write(json_file_path, row.to_json)
  end
end
